src/Service/Controllers/RestController.cs

```diff
@@ -224,16 +224,21 @@ public static JsonResult ErrorResponse(string code, string message, HttpStatusCo
 
                 if (result is CreatedResult)
                 {
-                    // Location is made up of two parts, the first being constructed
-                    // from the HttpRequest found in the HttpContext. The other part
-                    // is the primary key route, which has already been saved in the
+                    // Location is made up of three parts, the first being constructed
+                    // from the Host property found in the HttpContext.Request. The second part being the
+                    // base route configured in the config file.
+                    // The third part is the primary key route, which has already been saved in the
                     // Location of the created result. So we form the entire location
-                    // from appending the primary key route  already stored in the
+                    // from appending the base route and the primary key route  already stored in the
                     // created result to the url constructed from the HttpRequest. We
                     // then update the Location of the created result to this value.
                     CreatedResult createdResult = (result as CreatedResult)!;
-                    string location = UriHelper.GetEncodedUrl(HttpContext.Request) + "/" + createdResult.Location;
-                    createdResult.Location = location;
+                    string baseURL = UriHelper.BuildAbsolute(
+                                        scheme: HttpContext.Request.Scheme,
+                                        host: HttpContext.Request.Host,
+                                        pathBase: _restService.GetBaseRouteFromConfig(),
+                                        path: HttpContext.Request.Path);
+                    createdResult.Location = baseURL.EndsWith('/') ? baseURL + createdResult.Location : baseURL + "/" + createdResult.Location;
                     result = createdResult;
```

Comment:
Instead of string concatenation and ternary conditional operator `? :`, could you use the Microsoft.AspNetCore.Http `PathString.Add(PathString)` method? perhaps that will accommodate checking for trailing slashses. https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.http.pathstring.add?view=aspnetcore-6.0#microsoft-aspnetcore-http-pathstring-add(microsoft-aspnetcore-http-pathstring)

Reply:
Thanks for the suggestion. 

`PathString` expects the string to start with `/`, failing which it throws an exception. The `createdResult.Location` that we create does not have a leading `/`. To be able to use `PathString`, we need to add a leading slash to the PK part (createdResult.Location). AFAICT, manual check/addition cannot be avoided completely. 

Considering this, the current implementation should be fine - performs a check and introduces a `/` only when necessary